tests/vm_test.go

```diff
@@ -623,6 +623,31 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				Expect(err.Error()).To(Equal("Error stopping VirtualMachine VM is not running"))
 			})
 
+			It("[test_id:3007]Should force restart a VM with terminationGracePeriodSeconds>0", func() {
+				By("getting a VM with high TerminationGracePeriod")
+				newVMI := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskFedora))
+				gracePeriod := int64(1800)
+				newVMI.Spec.TerminationGracePeriodSeconds = &gracePeriod
+				newVM := tests.NewRandomVirtualMachine(newVMI, true)
+				_, err := virtClient.VirtualMachine(newVM.Namespace).Create(newVM)
+				Expect(err).ToNot(HaveOccurred())
+				waitForVMIStart(virtClient, newVMI)
+
+				oldCreationTime := newVMI.ObjectMeta.CreationTimestamp
+				oldVMIUuid := newVM.ObjectMeta.UID
+
+				By("Invoking virtctl --force restart")
+				forceRestart := tests.NewRepeatableVirtctlCommand(vm.COMMAND_RESTART, "--namespace", newVM.Namespace, "--force", newVM.Name)
+				err = forceRestart()
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Comparing the new CreationTimeStamp with the old one")
```

Comment:
Even if you force-delete a pod, it will respect for instance finalizers. I don't think that you can assume that the VMI immediately disappeared. I think that this will be very flaky.

Reply:
Right, this can be flaky, added the waitForScheduling to make sure there's a new VMI in place